src/impl/pooling_direct.jl

```diff
@@ -13,13 +13,13 @@ for name in (:max, :mean, :lpnorm)
     end
 
     @eval function $((Symbol("$(name)pool_direct!")))(
-        y::AbstractArray{T,5}, x::AbstractArray{T,5},
+        y::AbstractArray{T,5}, x::AbstractArray{R,5},
```

Comment:
```suggestion
        y::AbstractArray{T,5}, x::AbstractArray{<:Any,5},
```
Any type parameters not used in the function should be unnecessary